typescript/web/src/components/auth-manager/signin-modal/signin-modal.tsx

```diff
@@ -161,8 +167,25 @@ export const SigninModal = ({
                     </>
                   ) : (
                     <>
-                      <FormControl id="email">
+                      <FormControl
+                        id="email"
+                        isRequired
+                        isInvalid={isEmailInvalid}
+                      >
                         <Input
+                          onChange={(event) => {
+                            setIsEmailInvalid(
+                              !validateEmail(event.target.value)
+                            );
+                          }}
+                          onBlur={(event) => {
+                            if (event.target.value.length === 0) {
```

Comment:
If you add the check on `event.target.value.length === 0` above then I don't think you need this

Reply:
It is still needed otherwise the border always remains red after being blurred out. I think we want it to go back to the default state when it's empty